### ISSUE TITLE
Lint/AmbiguousBlockAssociation - a false positive case for proc

### ISSUE THREAD

**Issue by deepj** [rubocop/rubocop#10566]
## Expected behavior

The following code is used in a Capistrano script.

```ruby
ask :branch, proc { `git rev-parse --abbrev-ref HEAD`.chomp }
```

The code doesn't violate `Lint/AmbiguousBlockAssociation` cop

## Actual behavior

```
W: Lint/AmbiguousBlockAssociation: Parenthesize the param proc { git rev-parse --abbrev-ref HEAD.chomp } to make sure that the block will be associated with the proc method call.
```
When I use `lambda` or `->` instead of `proc` or `Proc.new`, this cop is not violated.

One note: ` characters from the block are omitted in the cop's message

## Steps to reproduce the problem

Code snippet

```ruby
ask :branch, proc { `git rev-parse --abbrev-ref HEAD`.chomp }
```

Run

```
rubocop --only Lint/AmbiguousBlockAssociation
```

## RuboCop version

```shell
$ rubocop -V
1.28.1 (using Parser 3.1.2.0, rubocop-ast 1.17.0, running on ruby 3.1.2 arm64-darwin21)
  - rubocop-performance 1.13.3
  - rubocop-rails 2.14.2
```

**Comment by TylerRick** [comment]
Nit pick: That's actually a **false negative** if _doesn't_ trigger the cop but _should_; a **false positive** is when it _does_ trigger the cop/test/fire alarm/etc. but _should not_.
